test/controllers/drivers_controller_test.rb

```diff
@@ -55,121 +51,183 @@
       # Arrange
       # Ensure there is an invalid id that points to no driver
-
-      # Act
-
-      # Assert
+      get edit_driver_path(-1)
+      # Act/ssert
+      must_respond_with :redirect
 
     end
   end
 
   describe "update" do
+    before do
+      Driver.create(
+        name: "Jane Doe",
+        vin: "ww234444112345652",
+        available: true )
+    end
+
+    let (:update_driver_hash) do
+      {
+        driver:{
+          name: "John Doe",
+          vin: "ww234444112345651",
+          available: false
+        }
+      }
+    end
+
     it "can update an existing driver with valid information accurately, and redirect" do
-      # Arrange
-      # Ensure there is an existing driver saved
-      # Assign the existing driver's id to a local variable
-      # Set up the form data
+      driver_id = Driver.first.id
+      expect {
+        patch driver_path(driver_id), params: update_driver_hash
+      }.wont_change "Driver.count"
 
-      # Act-Assert
-      # Ensure that there is no change in Driver.count
+      must_respond_with :redirect
 
-      # Assert
-      # Use the local variable of an existing driver's id to find the driver again, and check that its attributes are updated
-      # Check that the controller redirected the user
+      driver = Driver.find_by(id: driver_id)
+      expect(driver.name).must_equal update_driver_hash[:driver][:name]
+      expect(driver.vin).must_equal update_driver_hash[:driver][:vin]
+      expect(driver.available).must_equal update_driver_hash[:driver][:available]
 
     end
 
-    it "does not update any driver if given an invalid id, and responds with a 404" do
+    it "does not update any driver if given an invalid id, and redirect to drivers path" do
       # Arrange
       # Ensure there is an invalid id that points to no driver
       # Set up the form data
-
       # Act-Assert
       # Ensure that there is no change in Driver.count
-
       # Assert
       # Check that the controller gave back a 404
 
+      expect {
+        get edit_driver_path(-1), params: update_driver_hash
+      }.wont_change "Driver.count"
+      must_redirect_to drivers_path
+
     end
 
     it "does not create a driver if the form data violates Driver validations, and responds with a redirect" do
-      # Note: This will not pass until ActiveRecord Validations lesson
-      # Arrange
-      # Ensure there is an existing driver saved
-      # Assign the existing driver's id to a local variable
-      # Set up the form data so that it violates Driver validations
+
+      driver_hash = {
+        driver: {
+          name: "John Do",
+            vin: "ww234",
+            available: false
+        },
+      }
 
       # Act-Assert
       # Ensure that there is no change in Driver.count
+      expect {
+        post drivers_path, params: driver_hash
+      }.wont_change "Driver.count"
 
-      # Assert
-      # Check that the controller redirects
+      assert_template :new
```

Comment:
You don't need to test which template shows (this is depreciated in Rails nowadays).  

You **should** check the status code.